autowarefoundation/autoware
# Predicting convex hull in naive_motion_predict

## New feature implementation

### Implemented feature

Related issue: autowarefoundation/autoware_ai#642 


![image](https://user-images.githubusercontent.com/10920881/57745378-e3b0c500-7707-11e9-8118-bc0adfcd3cbe.png)


### Implementation description


Current implementation in `naive_motion_predictit` doesn't predict convex hull in `DetectedObjectArray`.
This PR aims to add function for predicting convex hull in `naive_motion_predict` package.

### How to test new feature

1. Download a rosbag for the test from [here](https://drive.google.com/file/d/1cUDUQ8Qyn_Zmvt1mOX0O58w3WiFfYbk7/view?usp=sharing).
2. First, check the shape of the convex hull by using `costmap_generator`.
2-a. Open runtime manager and play the rosbag.
2-b. Click `app` next to `costmap_generator` in `Computing` tab.
![image](https://user-images.githubusercontent.com/10920881/57745811-8ddd1c80-7709-11e9-8fc7-5d3de80b7dce.png)
2-c. Click `use_objects_convex_hull`, Change `input_topic` to `/detection/objects`, change `expand_polygon_size` to `0`, change `size_of_expansion_kernel` to `1`.
2-d.  Launch `costmap_generator`.
2-e. Open rviz and check the subscription to `/semantics/costmap_generator/occupancy_grid`
![image](https://user-images.githubusercontent.com/10920881/57745919-0e038200-770a-11e9-864b-e0e2ce552c7e.png)
2-f. You will see hexagon shape.
![image](https://user-images.githubusercontent.com/10920881/57746024-73f00980-770a-11e9-8e15-4feb0b8013a4.png)
3. Next check the predicted convex hull.
3.a. Back to run time manager and launch `naive_motion_predict` in `Computing` tab.
3.b. Click `app` next to `costmap_generator`.
3.c. **Change `input_topic` to `/prediction/motion_predictor/objects`.**
3.d. Launch `costmap_generator`, check rviz.
3.e. You will see predicted convex hull.
![image](https://user-images.githubusercontent.com/10920881/57746234-4e173480-770b-11e9-8df8-c8410f000dc9.png)
